Remote clusters missing use cases and structural changes (#4170)
elastic/docs-content: deploy-manage/remote-clusters/eck-remote-clusters-from-external.md

Reviewer: needs introduction from https://docs-v3-preview.elastic.dev/elastic/docs-content/pull/4170/deploy-manage/remote-clusters/remote-clusters-self-managed

^^ that page also needs to link to the ECK section (it currently links to ech and ece in the intro)
<img width="744" height="170" alt="image" src="https://github.com/user-attachments/assets/ca78ae33-27ed-44f7-bd63-a8d7929f15db" />

:::{include} _snippets/eck_rcs_retrieve_ca.md
:::

#### Create a cross-cluster API key on the remote cluster

:::{include} _snippets/apikeys-create-key.md
:::

#### Configure the local deployment [configure-local-cluster]

:::{include} _snippets/apikeys-local-config-intro.md
:::

:::{include} _snippets/self_rcs_local_config.md
:::

::::::

::::::{tab-item} TLS certificate (deprecated)

:::{include} _snippets/tlscerts-intro.md
:::

#### Make sure both clusters trust each other's certificate authority [mtls-setup-eck]

:::{include} _snippets/eck_rcs_mtls_intro.md
:::

In this example:

* The remote cluster resides inside Kubernetes and is managed by ECK
* The local cluster is a self-managed cluster running outside of Kubernetes.

To allow mutual TLS authentication between the clusters:

1. Extract and save the transport CAs of both clusters:

    1. For the ECK-managed cluster:

        :::{include} _snippets/eck_rcs_certs_retrieve_ca.md
        :::

    2. For the self-managed cluster (self-managed CA pending) take the transport CA from any of the nodes of the cluster. You can save it as `self-managed_ca.crt`.

> 2. Configure the self-managed cluster to trust the transport CA of the ECK cluster, by adding the CA to the list of CAs in [`xpack.security.transport.ssl.certificate_authorities`](elasticsearch://reference/elasticsearch/configuration-reference/security-settings.md#_pem_encoded_files_3).

Reviewer: > by adding the CA

which ca? to which cluster? 

I can tell which (it's implied), but it's a thinker

> 3. Configure the ECK managed cluster to trust the transport CA of the self-managed cluster:

Reviewer: ```suggestion
3. Configure the ECK-managed cluster to trust the transport CA of the self-managed cluster:
```

---
navigation_title: From a self-managed cluster
applies_to:
  deployment:
    eck: ga
products:
  - id: cloud-kubernetes
sub:
  local_type_generic: cluster
  remote_type_generic: cluster
  remote_type: ECK-managed
---

> # Connect a self-managed {{es}} cluster to an ECK-managed cluster [self-to-eck-remote-clusters]

Author: I was planning to move it eventually to self-managed in the TOC at a later stage, as self-managed remote clusters docs haven't been refined or touched at all yet.

Maybe you can help me with a loose end that I left indirectly related with this.... (we can discuss in private also)

The seed of this doc was originally the section called `Connect from an Elasticsearch cluster running outside the Kubernetes cluster` in the original / unique doc. That original doc only covered TLS-based auth for self-managed or an external ECK towards ECK.

The content now is 95% applicable to sefl-managed -> ECK only, except the TLS based authentication (deprecated) where we consider that the remote could be managed by a different ECK operator (this is a small flaw I've left because I wasn't finding a better solution).

inter-ECKs remote clusters is covered for API key based auth in the new `to external` doc, but I didn't want to implement in that doc the TLS-based auth (as it would become very complex), so TLS based auth is covered for inter-ECKs is covered here without highlighting it much (it's probably not a good practice to implement it considering we now have API key based auth).

> # Connect a self-managed {{es}} cluster to an ECK-managed cluster [self-to-eck-remote-clusters]

Author: @shainaraskas , I think I now have a good plan to solve what is concerning me in a much nicer way, I'll share something on Monday. In short:

1) I'll move this doc to `self-managed` section, and it will be focused **only** in self-managed --> ECK (without that exception I shared that considered inter-ECKs conf).

2) I'll create a new doc "Connect Elasticsearch clusters in different ECK environments" where we will exclusively focus on that use case. IMO this is probably a common use case, more than mixing deployment types, and it deserves such a doc. It would be also similar to what we have with ECE and ECH when we talk about same ECE / different ECE or same ORG / different ORG.

3) The "Connect to external clusters" will be focused on connecting to either ECE/ECH or self-managed, and I'll remove a different ECK from that doc, as that will be covered in the previous doc.

> # Connect a self-managed {{es}} cluster to an ECK-managed cluster [self-to-eck-remote-clusters]

Author: done

:::{include} _snippets/apikeys-local-config-intro.md
:::

:::{include} _snippets/self_rcs_local_config.md
:::

::::::

::::::{tab-item} TLS certificate (deprecated)

#### Make sure both clusters trust each other’s certificate authority [k8s_make_sure_both_clusters_trust_each_others_certificate_authority]

When using TLS certificate–based authentication, the first step is to establish mutual trust between the clusters at the transport layer. This requires exchanging and trusting each cluster’s transport certificate authority (CA):
* The CA of the remote (ECK-managed) cluster must be added as a trusted CA in the local cluster,
* The local cluster’s transport CA must be added as a trusted CA in the remote cluster.

::::{note}
While it is technically possible to configure remote cluster connections using earlier versions of {{es}}, this guide only covers the setup for {{es}} 7.6 and later. The setup process is significantly simplified in {{es}} 7.6 due to improved support for the indirection of Kubernetes services.
::::

Consider the following example:

* `remote-cluster` resides inside Kubernetes and is managed by ECK

> * `local-cluster` is not hosted inside the same Kubernetes cluster as `remote-cluster` and might not even be managed by ECK

Reviewer: can we refine this sentence for its context

To allow mutual TLS authentication between the clusters:

1. The certificate authority (CA) used by ECK to issue certificates for the {{es}} transport layer is stored in a secret named `<cluster_name>-es-transport-certs-public`. Extract the certificate for `remote-cluster` as follows:

    ```sh
    kubectl get secret remote-cluster-es-transport-certs-public \
    -o go-template='{{index .data "ca.crt" | base64decode}}' > remote.ca.crt
    ```

    ::::{note}

>     Beware of copying the source secret as-is into a different namespace. Check [Common Problems: Owner References](../../troubleshoot/deployments/cloud-on-k8s/common-problems.md#k8s-common-problems-owner-refs) for more information.

Reviewer: ```suggestion
    Beware of copying the source secret as-is into a different namespace. Refer to [Copying secrets with Owner References](../../troubleshoot/deployments/cloud-on-k8s/common-problems.md#k8s-common-problems-owner-refs) for more information.
```
